### DISCUSSION TITLE
Defer execution of mutationStore in @urql/svelte

### DISCUSSION THREAD

**Discussion by SiNONiMiTY** [urql-graphql/urql#3105] [Q&A] [answered]
Title says, how to defer/manually execute mutationStore? As soon as the page loads, it is executed and a blank request is sent to the endpoint.

**Comment by kitten** [comment, marked as answer]
The discussion issue you commented on (#2499) assumes that you cannot invoke and create a store dynamically, e.g. in a callback, which you can, so the following is possible:

```svelte
<script>
  import { readable } from 'svelte/store';
	
  let result;
  let client = null; // getContextClient();
  const updateTodo = () => {
    // use `mutationStore()` here
    result = readable(null, set => {
      set({ fetching: true, data: undefined });
      const id = setTimeout(() => {
        set({ fetching: false, data: { setTitle: 'test' } });
      }, 1000);
      return () => clearInterval(id);
    });
  };
</script>

{#if $result}
  {#if $result.fetching}
    <small>loading</small>
  {:else}
    <p>Updated {$result.data.setTitle}!</p>
  {/if}
{:else}
  <small>[inactive]</small>
  <button type="button" on:click={updateTodo}>
    Execute Mutation
  </button>
{/if}
```

You can paste this into the [Svelte REPL](https://svelte.dev/repl/) to test it.

Basically, if you invoke `mutationStore` in a function then it won't be created and won't start immediately and you will still get a store as a result, which you can assign to a local variable of your component.

The alternative is to use `client.mutation(...).toPromise` (which is pointed out in other discussions or issues), but if you need a store, the best way is to actually just create it on the fly.

**Reply by SiNONiMiTY** [reply]
Thanks for the explanation and example @kitten. I will try to use the client.mutation() API and see if it works in my case rather than forcing mutations to the stores.
